dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -2906,15 +2906,36 @@ SimpleColumn* getSmallestColumn(boost::shared_ptr<CalpontSystemCatalog> csc,
 
             if (tan.alias == csep->derivedTbAlias())
             {
-                assert (!csep->returnedCols().empty());
-                ReturnedColumn* rc = dynamic_cast<ReturnedColumn*>(csep->returnedCols()[0].get());
+                const CalpontSelectExecutionPlan::ReturnedColumnList& cols = csep->returnedCols();
+
+                assert (!cols.empty());
```

Comment:
I hope that we will run MTR using debug builds otherwise this assert effectively does nothing. IMHO better to check cols for emptiness before col[0] will be referenced in the line 2928.

Reply:
Good catch. I will move the assert later where we reference cols[0].